server/routes/charts.js

```diff
@@ -1,32 +1,73 @@
 import { Router } from 'express';
 
 import ensureAuthenticated from '../utils/passport/ensure-authenticated';
+import { collections } from '../utils/mongoCollections'
 
 import chartsListPage from '../templates/Chart.template'
 import newChartPage from '../templates/NewChart.template'
 
 const router = Router();
 
 router.route('/charts')
-  .get(ensureAuthenticated, async (_, res) => {
+  .get(ensureAuthenticated, async (req, res) => {
     try {
       return res.status(200).send(chartsListPage())
-    } catch (err) {
-      console.error(err.message)
+    } catch (error) {
+      console.error(error.message)
 
       return res.status(500).send({ message: err.message })
     }
   });
   
 router.route('/charts/new')
-  .get(ensureAuthenticated, async (_, res) => {
+  .get(ensureAuthenticated, async (req, res) => {
     try {
-      return res.status(200).send(newChartPage())
+      const { display_name, role, icon } = req.session;
+      const user = {
+        uid: req.user,
+        name: display_name,
+        role, 
+        icon,
+      };
```

Comment:
since we do this in a number of places, what do you think about extracting the logic to a helper/util?

```suggestion
      const user = userFromRequest(req)
```

Reply:
[8a7c3d0](https://github.com/AMP-SCZ/dpdash/pull/84/commits/8a7c3d014d0266427e76ae9b5d15d0618767ac3a)